tests/Testcontainers.Tests/Unit/Containers/Unix/ProtectDockerDaemonSocketTest.cs

```diff
@@ -0,0 +1,84 @@
+namespace DotNet.Testcontainers.Tests.Unit.Containers.Unix
+{
+  using System;
+  using System.Collections.Generic;
+  using System.IO;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Builders;
+  using DotNet.Testcontainers.Clients;
+  using DotNet.Testcontainers.Configurations;
+  using DotNet.Testcontainers.Tests.Fixtures;
+  using Microsoft.Extensions.Logging.Abstractions;
+  using Xunit;
+
+  public static class ProtectDockerDaemonSocketTest
+  {
+    private static ICustomConfiguration GetCustomConfiguration(ProtectDockerDaemonSocket protectDockerDaemonSocket, bool tlsEnabled, bool tlsVerifyEnabled)
```

Comment:
No matter which combination I use for `tlsEnabled` and `tlsVerifyEnabled` the tests always succeed. I would expect a different behaviour here.

Reply:
I will check but be carefull what the entrypoint.sh script of the docker image really does. I failed several times until I found that --tlsverify=false switch

Reply:
Haha, that was nice catch. I removed these flags because they are useless in commit 2e11eae.
These flags are used only in method `bool IDockerEndpointAuthenticationProvider.IsApplicable()` to ensure if this provider should be used when is registered in collection `TestcontainersSettings.DockerEndpointAuthConfig`. This collection is not used by tests.

Reply:
Makes sense 👍. I think we can use it to test `IsApplicable` too in the future. Something like:

```csharp
new[] { MTlsEndpointAuthenticationProvider(customConfiguration), TlsEndpointAuthenticationProvider(customConfiguration) }.First(...)
Assert.IsType<MTlsEndpointAuthenticationProvider>(...)
```

Reply:
IsApplicable is already tested by test `DockerEndpointAuthenticationProviderTest.AuthProviderShouldBeApplicable`. I think it's all.